### PR TITLE
TB-3634 Sources analytics events

### PR DESCRIPTION
**Pull request by frankpepermans**
### What 🕵️ 🔍

- Adds 2 new analytics events to track changes in regards to source management

----------

### How to test (please adjust template) 🥼 🔬
- Feature flag disabled:
  - [x] From the feed, dislike a card, then choose to add the source of the card as "excluded source". Expect that an event fires in Analytics. (`sourceChanged`)
  - [x] From the settings > sources: do some changes (add/remove) on both lists, when finally closing this management screen, expect an Analytics event to fire (`sourcesChanged`)
- [x] Finally, also expect events to trigger whenever screens are opened (`sourceFeedSetting`, `excludedSourceSelection`, `trustedSourceSelection`)
----------

### References 📝 🔗

- [JIRA](https://xainag.atlassian.net/browse/TB-XXXX)

----------

### CODE REVIEW

File: application/lib/infrastructure/service/analytics/events/sources_management_changed_event.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    required int oldTrustedCount,
+    required int newExcludedCount,
+    required int oldExcludedCount,
+  }) : super(
+          _kEventType,
+          properties: {
+            _kParamNewTrustedCount: newTrustedCount,
+            _kParamOldTrustedCount: oldTrustedCount,
+            _kParamNewExcludedCount: newExcludedCount,
+            _kParamOldExcludedCount: oldExcludedCount,
```
[the remark this answers is not in the document]
Comment (frankpepermans) [reply application/lib/infrastructure/service/analytics/events/sources_management_changed_event.dart:21]: The truth is in the engine, they store the data, so  when we open the settings, it fetches data from the engine, the event then just logs differences

**Comment by shushanAspaturyan** [comment]
@frankpepermans those events are not firing - sourceFeedSetting, excludedSourceSelection, trustedSourceSelection.
Also sourceChanged is not firing after you delete all the sources together.

**Comment by frankpepermans** [comment]
@shushanAspaturyan are you checking on mixpanel

File: application/lib/infrastructure/service/analytics/events/sources_management_changed_event.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichaelKMalak) [reply application/lib/infrastructure/service/analytics/events/sources_management_changed_event.dart:21]: yeah I mean, a user may just exclude sources and not open the settings for a week or so.. 
It'd be nice to update user profile with the number like we already do with # bookmarks, # collection, # active countries. But, it doesn't need to be part of this PR ofc, it's just a suggestion.

**Review by MichaelKMalak — APPROVED** [review]

File: application/lib/infrastructure/service/analytics/events/sources_management_changed_event.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichaelKMalak) [reply application/lib/infrastructure/service/analytics/events/sources_management_changed_event.dart:21]: I see.. makes sense 👍

**Comment by shushanAspaturyan** [comment]
> @shushanAspaturyan they are openScreen events, like all others, they contain the values described

ah ok! now I see them. thanks for clarification

**Comment by shushanAspaturyan** [comment]
> @shushanAspaturyan are you checking on mixpanel

yup

**Comment by shushanAspaturyan** [comment]
>>Also sourceChanged is not firing after you delete all the sources together.
Can not reproduce this issue

**Pull request merged**